From 1d1604561583c26f9297b3f7e3942b9c1d078b19 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:19:55 -0400
Subject: [alsa] hda: Remove dependency on bus->pci in hda_beep.c

Message-id: <140972823573-122-patchset-jkysela@redhat.com>
Patchwork-id: 90822
O-Subject: [RHEL 7.1 PATCH 121/442] ALSA: hda - Remove dependency on bus->pci in hda_beep.c
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The default parent device can be obtained directly via card object, so
we don't need to rely on pci->dev.parent.  Since there is no access to
pci_dev, we can reduce the inclusion of linux/pci.h, too.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: ed9d0b626e65ab653e5421083d8dcd59cacd1132)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_beep.c b/sound/pci/hda/hda_beep.c
index 0589b39..d8c437a 100644
--- a/sound/pci/hda/hda_beep.c
+++ b/sound/pci/hda/hda_beep.c
@@ -20,7 +20,6 @@
  */
 
 #include <linux/input.h>
-#include <linux/pci.h>
 #include <linux/slab.h>
 #include <linux/workqueue.h>
 #include <linux/export.h>
@@ -167,7 +166,7 @@ static int snd_hda_do_attach(struct hda_beep *beep)
  input_dev->evbit[0] = BIT_MASK(EV_SND);
  input_dev->sndbit[0] = BIT_MASK(SND_BELL) | BIT_MASK(SND_TONE);
  input_dev->event = snd_hda_beep_event;
- input_dev->dev.parent = &codec->bus->pci->dev;
+ input_dev->dev.parent = codec->bus->card->dev;
  input_set_drvdata(input_dev, beep);
 
  err = input_register_device(input_dev);
-- 
1.7.1